woocommerce/woocommerce-android
# issue/4676 Implemented support for update cancelation

Closes: #4676 

### Description
The PR adds functionality of cancelation of the required and optional updates

### Testing instructions
*Required*
* Change `TerminalWarpper::77` to `REQUIRED`
* Start collecting payment
* Notice that required update starts
* Try to cancel it with the `cancel` button
* Try to collect payment again
* Notice that update starts again
* Try the same from the Settings -> Mange card reader menu

*Optional*
* Change `TerminalWarpper::77` to `UPDATE_AVAILABLE`
* Open Settings -> Mange card reader menu
* Start connecting to a reader
* Notice Update button is visible
* Start update
* Cancel
* Try to update again

### Images/gif

https://user-images.githubusercontent.com/4923871/132466172-e0c4ef1f-6839-45ef-b0e6-adedd89c400d.mp4

https://user-images.githubusercontent.com/4923871/132466213-2212dfe6-5024-41f0-a2e0-01a3c7bc8d78.mp4
